testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.test.stabilitylevel;
+
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.ClientConstants;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.integration.management.ManagementOperations;
+import org.jboss.as.test.integration.management.util.ServerReload;
+import org.jboss.as.version.Stability;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Assume;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE_SERVICE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.NAME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.READ_OPERATION_DESCRIPTION_OPERATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.READ_OPERATION_NAMES_OPERATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RELOAD_ENHANCED;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.STABILITY;
+import static org.jboss.as.server.controller.descriptions.ServerDescriptionConstants.SERVER_ENVIRONMENT;
+import static org.junit.Assert.fail;
+
+/**
+ * For tests that need to run under a specific server stability level,
+ * the server setup tasks from the inner classes can be used to change the stability level of the server to the desired level.
+ * Once the test is done, the original stability level is restored.
+ *
+ * In order to not pollute the configuration with XML from a different stability level following the run of the test,
+ * it takes a snapshot of the server configuration in the setup() method, and
+ */
+public abstract class StabilityServerSetupSnapshotRestoreTasks implements ServerSetupTask {
+
+    private final Stability desiredStability;
+    private volatile Stability originalStability;
+
+    private AutoCloseable snapshot;
+
+
+    public StabilityServerSetupSnapshotRestoreTasks(Stability desiredStability) {
+        this.desiredStability = desiredStability;
+    }
+
+    @Override
+    public final void setup(ManagementClient managementClient, String containerId) throws Exception {
+        // Make sure the desired stability level is one of the ones supported by the server
+        Set<Stability> supportedStabilityLevels = getSupportedStabilityLevels(managementClient);
+        Assume.assumeTrue(
+                String.format("%s is not a supported stability level", desiredStability, supportedStabilityLevels),
```

Comment:
This is not a valid statement -- two vars but only one expression in the format string